name: bug_tracker
services:
  db:
    image: postgres:17
    restart: always
    volumes:
      - db-data:/var/lib/postgresql/data
    environment:
      - POSTGRES_DB=${DATABASE_NAME}
      - POSTGRES_USER=${DATABASE_USER}
      - POSTGRES_PASSWORD=${DATABASE_PASSWORD}
      - POSTGRES_SERVER=${DATABASE_SERVER}
      - PGDATA=/var/lib/postgresql/data
    ports:
      - "5432:5432"
    networks:
      - app_network
  web:
    build:
      context: .
      dockerfile: ./docker/django/Dockerfile
      args:
        INSTALL_DEV: true
    working_dir: /app/bug_tracker
    command: >
      bash -c "python manage.py wait_for_db &&
               python manage.py migrate &&
               python manage.py runserver 0.0.0.0:8000"
    volumes:
      - .:/app
    expose:
      - 8000
    env_file:
      - .env
    depends_on:
      - db
    networks:
    - app_network
  nginx:
    build: ./docker/nginx
    volumes:
      - ./docker/nginx/nginx.conf:/etc/nginx/conf.d/nginx.conf
      - ./media:/app/media
    ports:
      - "80:80"
    depends_on:
      - web
    networks:
      - app_network
  redis:
    image: redis:7-alpine
    restart: always
    ports:
      - "6379:6379"
    volumes:
      - redis-data:/data
    networks:
      - app_network
  celery:
    build:
      context: .
      dockerfile: ./docker/django/Dockerfile
    command: celery -A bug_tracker worker --loglevel INFO
    working_dir: /app/bug_tracker
    depends_on:
      - redis
      - web
    volumes:
      - .:/app
    networks:
      - app_network
  flower:
    build:
      context: .
      dockerfile: ./docker/django/Dockerfile
    working_dir: /app/bug_tracker
    command: celery -A bug_tracker flower --port=5555 --loglevel INFO
    depends_on:
      - redis
      - celery
    ports:
      - "5555:5555"
    networks:
      - app_network
volumes:
  db-data:
  redis-data:
networks:
  app_network:
    driver: bridge
